hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java

```diff
@@ -753,7 +758,6 @@ public EntityFetch generateFetch(
 			return new EntityFetchSelectImpl(
 					fetchParent,
 					this,
-					isNullable,
```

Comment:
We also need to record whether a foreign key constraint is available on the respective ForeignKey instance and based on that and the `ignoreNotFound` flag, adapt `EntityFetchJoinedImpl` to also create a domain result for the foreign key part to be able to check inconsistencies.

Reply:
yeap